src/Illuminate/Database/Eloquent/Concerns/QueriesRelationships.php

```diff
@@ -470,6 +471,17 @@ public function orWhereMorphedTo($relation, $model)
      */
     public function whereBelongsTo($related, $relationshipName = null, $boolean = 'and')
     {
+        if (! $related instanceof Collection) {
+            $values = $related->newCollection([$related]);
+        } else {
+            $values = $related;
+            $related = $values->first();
```

Comment:
What happens if you pass in an empty collection? I would expect to get no models back after the query is run.